### PR TITLE
SDI-138 Add endpoints into csr-api

### PR DESCRIPTION
**Pull request by steverendell**
Tripped over the limit of 1000 on oracle "in (..)" clause,
Also use path variables for region number in new endpoints

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -37,8 +37,6 @@ dependencies {
   implementation("org.springdoc:springdoc-openapi-security:1.6.6")
 
   implementation("org.apache.commons:commons-lang3:3.12.0")
-  implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.13.2")
-  implementation("com.fasterxml.jackson.module:jackson-module-kotlin:2.13.2")
```
Comment (steverendell) [inline build.gradle.kts:41]: not explicitly needed

File: src/test/resources/application-test.yml [hunk 2]
```diff
@@ -36,7 +36,13 @@ api:
 csr:
   regions:
     - name: 1
-      url: 'jdbc:h2:mem:csr-api-test;Mode=Oracle;'
+      url: 'jdbc:h2:mem:csr-api-test-1;Mode=Oracle'
+    #  url: 'jdbc:h2:file:C:\users\steve\git\csr-api\csr-api-test;Mode=Oracle'
```
Comment (steverendell) [inline src/test/resources/application-test.yml:40]: whoops - will remove next time

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt [hunk 3]
```diff
@@ -19,16 +22,20 @@ import uk.gov.justice.digital.hmpps.csr.api.dto.DetailDto
...
 class DetailController(private val detailService: DetailService) {
 
   @Operation(summary = "Retrieve all details for a user between two dates")
+  @Parameter(
+    `in` = ParameterIn.HEADER,
+    name = "X-Region",
+    description = "the number of the required region, 1-6",
+    required = true
+  )
   @GetMapping("/user/details")
```
Comment (lbennett-moj) [inline src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt:37]: why was this needed?

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailService.kt [hunk 4]
```diff
@@ -71,31 +86,46 @@ class DetailService(private val sqlRepository: SqlRepository, val authentication
...
     val startTime = System.currentTimeMillis()
-    val deleted = sqlRepository.deleteProcessed(ids)
-    log.info("deleteProcessed: received ${ids.size} ids, deleted $deleted rows, time taken ${elapsed(startTime)}s")
+    RegionContext.setRegion(region.toString())
+
+    ids.chunked(DELETECHUNKSIZE).forEach {
+      val deleted = sqlRepository.deleteProcessed(it)
+      log.debug("deleteProcessed: deleted $deleted rows")
+    }
+
```
Comment (lbennett-moj) [inline src/main/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailService.kt:108]: would you need to catch a failure here to allow it to continue?

**Review by lbennett-moj — APPROVED** [review]

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt (hunk 3 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt:37]: its not actually. I originally wanted to add the region description to highlight that it is needed for the manual admin endpoints too, but now Ive changed the region parameter to be on path. I guess its harmless tho
